diff --git a/imagecodecs/brunsli.pxd b/imagecodecs/brunsli.pxd
index 09593e2..0363e34 100644
--- a/imagecodecs/brunsli.pxd
+++ b/imagecodecs/brunsli.pxd
@@ -21,7 +21,7 @@ cdef extern from 'brunsli/decode.h':
     ) nogil
 
 
-cdef extern from 'brunsli/decode.h':
+cdef extern from 'brunsli/encode.h':
 
     int EncodeBrunsli(
         size_t size,
diff --git a/setup.py b/setup.py
index b3107c8..fecbcff 100644
--- a/setup.py
+++ b/setup.py
@@ -172,6 +172,8 @@ EXTENSIONS = {
     'zstd': ext(libraries=['zstd']),
 }
 
+EXTENSIONS_LINK = {}
+
 
 def customize_build_default(EXTENSIONS, OPTIONS):
     """Customize build for common platforms: recent Debian, arch..."""
@@ -266,6 +268,9 @@ def customize_build_ci(EXTENSIONS, OPTIONS):
     del EXTENSIONS['jpeg12']
 
     if not os.environ.get('SKIP_OMP', False):
+        if sys.platform == "darwin":
+            EXTENSIONS['zfp']['extra_compile_args'].append('-Xpreprocessor')
+            EXTENSIONS_LINK['zfp'] = ['-lomp']
         EXTENSIONS['zfp']['extra_compile_args'].append('-fopenmp')
 
     base_path = os.environ.get(
@@ -484,6 +489,7 @@ def extension(name):
         extra_compile_args=(
             OPTIONS['extra_compile_args'] + ext['extra_compile_args']
         ),
+        extra_link_args=EXTENSIONS_LINK.get(name, [])
     )
 
 
